GraphQL-Schema-Design.md

```diff
@@ -0,0 +1,142 @@
+## Notes
+
+* GraphQL exposes a data set as [a graph](https://en.wikipedia.org/wiki/Graph_theory). See
+  [Explaining GraphQL Connections](https://dev-blog.apollodata.com/explaining-graphql-connections-c48b7c3d6976) for a
+  more detailed overview on how this applies to our codebase.
+
+# How to model our graph
+
+* As with most GraphQL advice, our schema should strive to always be backwards compatible.
+  Thus careful consideration is needed about naming (be as unambiguous as possible) and the shape of the data returned
+  by a field.
+
+* Whenever a ‘namespace’ would be used in a field by adding a prefix or suffix (e.g. `for_sale_artwork` or
```

Comment:
Question here, should we maybe focus on the reusability of the namespace type, rather than the common name? To me, that's the bigger issue. It's pretty easy to imagine other things than a show having a location. Or to imagine heterogeneous nodes being aggregated by location. But it's less obvious that the name itself is a reason to aggregate.

Reply:
I think that’s a great addition! But the common name should not be overlooked, we wanna make sure to deliberately design the schema as a graph and not just 1:1 map backend data, even if that type were not to be re-used (or likely to).